src/FormBuilder/Routes/CreateFormRoute.php

```diff
@@ -50,4 +48,29 @@ public function __invoke()
             }
         }
     }
+
+
+    /**
+     * @unreleased
+     */
+    private function getFormSettings(): array
+    {
+        // Copy campaign goal settings
+        if (isset($_GET['campaignId'])) {
+            $campaign = Campaign::find((int)$_GET['campaignId']);
+
+            if ($campaign) {
+                return [
+                    'enableDonationGoal' => true,
+                    'goalAmount' => $campaign->goal,
+                    'goalType' => $campaign->goalType->getValue(),
+                ];
+            }
+        }
+
+        return [
+            'enableDonationGoal' => true,
+            'goalAmount' => 1000,
+        ];
+    }
```

Comment:
The best approach here would be to rollback these changes and implement this logic using hooks inside the `Campaigns` domain. You can use the `givewp_donation_form_creating` for that. 

As a sample, you can take a look at the `AddCampaignFormFromRequest` class which is triggered on the `givewp_donation_form_created` action.

Reply:
Another thing to consider is that we also need to use the `save_post_give_forms` hook to add this same logic for v2 forms. 

I used this same hook on #7571 to implement a way to add v2 forms to the campaigns. I did that by changing the `AddCampaignFormFromRequest` class to this: https://github.com/impress-org/givewp/blob/feature/feature-flag-GIVE-1264/src/Campaigns/Actions/AddCampaignFormFromRequest.php#L16-L26

Reply:
Resolved https://github.com/impress-org/givewp/pull/7569/commits/7acbe0937fcfd1bf7d67f005a807158f172aa187